### DISCUSSION TITLE
Resetable figure caption numbers

### DISCUSSION THREAD

**Discussion by yoonghm** [quarto-dev/quarto-cli#6767] [Q&A] [answered]
### Description

I would like to have a have a format with the following requirements:
1. Top-level headings are prefixed by a string, such as "Chapter 1", "Chapter 2", etc.
2. Figure and table numbers within a chapter is denoted using chapter number, and a running alphabet sequence, such as "Fig 1a", "Fig 1b" in "Chapter 1"; "Fig 2a", "Fig 2b" in "Chapter 2".

In Adobe Framemaker (FM), the auto-numbering format is handled using lists of integers See [https://help.adobe.com/en_US/framemaker/using/using-framemaker/user-guide/frm_page_layout_pl-styles-lists-and-autonumbering.html](https://help.adobe.com/en_US/framemaker/using/using-framemaker/user-guide/frm_page_layout_pl-styles-lists-and-autonumbering.html)

Let me illustrate here:

**Paragraph Style**|**Auto-numbering Format**
---------------------|--------------------------------
Heading 1              | `H: Chapter <n+>.< =0>< =0>< =0>`
Heading 2              | `H: <n>.<n+>< >< >`
Figure                    | `Figure: <n>< >-<a+>< >`
Table                      | `Table: <n>< >< ><A+>`

**Explanation**:

1. `<n+>` instructs FM to increment the numerical value by 1 whenever the paragraph format is encountered, and display the value.
2. `<n>` instructs FM to display the value.
3. `< =0>` instructs FM not to display the value, and reset the value to 0
4. `< >` instructs FM not to display the value.
5. `<a+>` instructs FM to increment the alphabet (in small capital) by 1, and display the value.
6. `<A+>` instructs FM to increment the alphabet (in big capital) by 1, and display the value.

So we would have the following auto-numbering strings:

Chapter 1
1.1
1.2
Figure 1-a
Figure 1-b
Table 1A
Table 1B
Figure 1.3

Chapter 2
Figure 2-a
Figure 2-b
2.1
2.2

**Comment by mcanouil** [comment]
I don't see any other solution than for you to implement that as a Lua Filter, see <https://quarto.org/docs/extensions/filters.html>.

**Reply by yoonghm** [reply]
Based on your suggestion on pandoc, I noticed that pandoc has provided such function but the format may not be compatible with markdown syntax. I would need to test ot out.

https://pandoc-numbering.readthedocs.io/en/latest/numbering.html

**Reply by yoonghm** [reply]
I have followed the steps in the pandoc documentation. It does not work. Quarto applies different parameters when invoking pandoc.

**Comment by cscheid** [comment, marked as answer]
I'd like to support this, but there are some of things blocking our ability to do that generically. See the second bullet item in https://github.com/quarto-dev/quarto-cli/issues/4944 if you want to track our progress on it.

**Reply by yoonghm** [reply]
Quarto is a good efforts in limiting the numbers of input and output document types. I am still new to Quarto and pandoc.

I am not sure if the following proposals are sensible and workable:

```
# Heading 1 Using Default Paragraph Style

A body text that uses default paragraph style
This line continues from previous line.

# Heading 1 using Special Paragraph Style {#ps-heading_style_1}

A body text that uses default paragraph style
This line continues from previous line.

Another body text that uses a customized paragraph style.
This line continues from previous line. {#ps-body_style_1}

## Heading 2 using default paragraph style

A body text that uses default paragraph style
This line continues from previous line.

## Heading 2 using a customized paragraph style {#ps-heading_style_2}

A body text that uses default paragraph style
This line continues from previous line.
```

`{#ps`, similar to `{#fig-` for figure caption, instruct Quarto to consider different paragraph style.

`heading_style_1`, `body_style_1`, `heading_style_2` are customized style which are defined per project or globally for several projects.

**Reply by cscheid** [reply]
Unfortunately it's not; we reserve `*-` prefixes for cross references.
